3rdparty/libprocess/src/tests/process_tests.cpp

```diff
@@ -1463,12 +1465,17 @@ TEST_F(ProcessTest, Remote)
   message.name = "handler";
   message.from = UPID("sender", sender.get());
   message.to = process.self();
+  message.body = "hello world";
 
   const string data = MessageEncoder::encode(message);
 
   AWAIT_READY(socket.send(data));
 
-  AWAIT_READY(handler);
+  AWAIT_READY(body);
+  ASSERT_EQ("hello world", body.get());
+
+  AWAIT_READY(pid);
+  ASSERT_EQ(message.from, pid.get());
```

Comment:
Would this assert fail if HOST header is not set with a real ip port?

Reply:
No it won't, it is just the pid of the sender process. Regardless of what the header is set this should succeed.

I actually wanted to write a Unit Test to verify host header being set at the receiver, however it is not straightforward to me from code to get host headers from libprocess' process. Message handlers in `process/process.hpp` seem to only care about process id and message body. I added the above as an Integration Test to verify connectivity/data transfer with changed header.